src/main/java/org/jabref/gui/actions/WriteXMPAction.java

```diff
@@ -116,49 +110,49 @@ private void writeXMP() {
                                     .map(Optional::get)
                                     .collect(Collectors.toList());
 
-            SwingUtilities.invokeLater(() -> optionsDialog.getProgressArea()
-                                                          .append(entry.getCiteKeyOptional().orElse(Localization.lang("undefined")) + "\n"));
+            Platform.runLater(() -> optionsDialog.getProgressArea()
```

Comment:
Shouldn't this already on the FX thread by defintion?

Reply:
No, `writeXMP` is invoked as a background task (but not properly implemented as `BackgroundTask`, leading to these ugly workaround - but I didnt have the motivation to implement it properly).